task/transcode.go

```diff
@@ -0,0 +1,237 @@
+package task
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"math/rand"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/livepeer/joy4/av"
+	"github.com/livepeer/joy4/av/avutil"
+	"github.com/livepeer/joy4/format"
+	"github.com/livepeer/joy4/format/mp4"
+	"github.com/livepeer/joy4/format/ts"
+	"github.com/livepeer/livepeer-data/pkg/data"
+	"github.com/livepeer/stream-tester/model"
+	"github.com/livepeer/stream-tester/segmenter"
+	"golang.org/x/sync/errgroup"
+)
+
+const (
+	segLen = 5 * time.Second
+)
+
+func init() {
+	format.RegisterAll()
+	rand.Seed(time.Now().UnixNano())
+}
+
+func TaskTranscode(tctx *TaskContext) (*data.TaskOutput, error) {
+	var (
+		ctx             = tctx.Context
+		inputPlaybackID = tctx.InputAsset.PlaybackID
+		lapi            = tctx.lapi
+	)
+
+	fullPath := videoFileName(inputPlaybackID)
+	fir, err := tctx.inputOS.ReadData(ctx, fullPath)
+	if err != nil {
+		return nil, fmt.Errorf("error reading data from source OS url=%s err=%w", fullPath, err)
+	}
+	fileInMem, err := io.ReadAll(fir.Body)
+	fir.Body.Close()
+	if err != nil {
+		return nil, err
+	}
+
+	streamName := fmt.Sprintf("vod_%s", time.Now().Format("2006-01-02T15:04:05Z07:00"))
+	profiles := tctx.Params.Transcode.Profiles
+	stream, err := lapi.CreateStreamEx(streamName, false, nil, profiles...)
```

Comment:
I think we might want to stick to a single `Profile`, at least on the first version, so that we don't need to make the `outputAssetId` an array instead which will be more complex to handle and index in the Postgres queries (we have queries to find the originating task from a given asset, for example). 

I remember that it is possible to index arrays in Postgres JSONs, but it required some different index type which we don't use yet and had some caveats, so we might to leave that for later. I think users will not be looking for multiple transcoding profiles on this API anyway, at least at first. Will probably make sense for some kind of ABR playback of an asset, in which case I'm not sure we will want to break up the result in multiple assets anyway.

Soo... WDYT of keeping the output a single `outputAssetId`, and either changing the task parameters to a single `Profile` or keeping it an array but checking both here and in the API that only 1 profile is present? That way the schema is a little more future-proof but we can make sure only 1 output asset is created right now.

Reply:
I've change code everywhere to use only one profile